src/util/apiUtils.js

```diff
@@ -201,56 +223,79 @@ export const apiResponseToQueryResponse = query => ({ value, meta }) => ({
 	}
 });
 
-/**
- * Parse request for queries and request options
- * @return {Object} { queries, externalRequestOpts }
- */
-export function parseRequest(request, baseUrl) {
-	const {
-		headers,
-		method,
-		query,
-		payload,
-		state,
-	} = request;
+export function getAuthHeaders({ state }) {
+	if (!state.MEETUP_MEMBER && state.oauth_token) {
+		return {
+			authorization: `Bearer ${state.oauth_token}`,
+		};
+	}
+	const cookies = { ...state };
+	const csrf = uuid.v4();
+	cookies.MEETUP_CSRF = csrf;
+	cookies.MEETUP_CSRF_DEV = csrf;
+	const cookie = Object.keys(cookies)
+		.map(name => `${name}=${cookies[name]}`).join('; ');
 
-	// Forward the Hapi request headers from the client query
-	// except for `host` and `accept-encoding`
-	// which should be provided by the external api request
+	return {
+		cookie,
+		'csrf-token': csrf,
+	};
+}
+
+export function parseRequestHeaders(request) {
 	const externalRequestHeaders = {
-		...headers,
-		authorization: `Bearer ${state.oauth_token}`,
+		...request.headers,
+		...getAuthHeaders(request),
 	};
 
-	delete externalRequestHeaders['host'];
-	delete externalRequestHeaders['accept-encoding'];
+	delete externalRequestHeaders['host'];  // let app server set 'host'
+	delete externalRequestHeaders['accept-encoding'];  // let app server set 'accept'
 	delete externalRequestHeaders['content-length'];  // original request content-length is irrelevant
-	delete externalRequestHeaders['cf-ray']; // cloudflare headers we don't want to pass on
+
+	// cloudflare headers we don't want to pass on
+	delete externalRequestHeaders['cf-ray'];
 	delete externalRequestHeaders['cf-ipcountry'];
 	delete externalRequestHeaders['cf-visitor'];
 	delete externalRequestHeaders['cf-connecting-ip'];
```

Comment:
probably - this code wasn't changed in this PR, though. I just added comments to make it more clear why each item was being deleted.

Definitely good for 2017 cleanup